hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/impl/GetTokenExpiryInfoPrecompile.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.store.contracts.precompile.impl;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateTrue;
+
+import com.hedera.services.context.primitives.StateView;
+import com.hedera.services.exceptions.InvalidTransactionException;
+import com.hedera.services.state.submerkle.ExpirableTxnRecord;
+import com.hedera.services.store.contracts.WorldLedgers;
+import com.hedera.services.store.contracts.precompile.SyntheticTxnFactory;
+import com.hedera.services.store.contracts.precompile.codec.DecodingFacade;
+import com.hedera.services.store.contracts.precompile.codec.EncodingFacade;
+import com.hedera.services.store.contracts.precompile.codec.TokenExpiryWrapper;
+import com.hedera.services.store.contracts.precompile.utils.PrecompilePricingUtils;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TransactionBody.Builder;
+import java.util.function.UnaryOperator;
+import org.apache.tuweni.bytes.Bytes;
+
+public class GetTokenExpiryInfoPrecompile extends AbstractReadOnlyPrecompile {
+    private final StateView stateView;
+
+    public GetTokenExpiryInfoPrecompile(
+            final TokenID tokenId,
+            final SyntheticTxnFactory syntheticTxnFactory,
+            final WorldLedgers ledgers,
+            final EncodingFacade encoder,
+            final DecodingFacade decoder,
+            final PrecompilePricingUtils pricingUtils,
+            final StateView stateView) {
+        super(tokenId, syntheticTxnFactory, ledgers, encoder, decoder, pricingUtils);
+        this.stateView = stateView;
+    }
+
+    @Override
+    public Builder body(final Bytes input, final UnaryOperator<byte[]> aliasResolver) {
+        final var getTokenExpiryInfoWrapper = decoder.decodeGetTokenExpiryInfo(input);
+        tokenId = getTokenExpiryInfoWrapper.tokenID();
+        return super.body(input, aliasResolver);
+    }
+
+    @Override
+    public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
+        validateTrue(stateView.tokenExists(tokenId), ResponseCodeEnum.INVALID_TOKEN_ID);
+        final var tokenInfo = stateView.infoForToken(tokenId).orElse(null);
+
+        if (tokenInfo == null) {
```

Comment:
You can use `validateTrue` check.

Reply:
Correct. In the initial implementation i have used validateTrueOrRevert, but for some reason sonar said that null pointer exception could be thrown because tokenInfo is nullable. So thats why i switched to normal if statement.